experimental/dds/tree/src/SharedTree.ts

```diff
@@ -545,7 +545,15 @@ export class SharedTree extends SharedObject<ISharedTreeEvents> implements NodeI
 		);
 
 		const attributionId = (options as SharedTreeOptions<WriteFormat.v0_1_1>).attributionId;
-		this.idCompressor = new IdCompressor(createSessionId(), reservedIdCount, attributionId, this.logger);
+		// Only emit compressor telemetry for 5% of sessions
+		const emitIdCompressorTelemetry = Math.random() < 0.05;
+		this.idCompressor = new IdCompressor(
+			createSessionId(),
+			reservedIdCount,
+			attributionId,
+			this.logger,
```

Comment:
Idea: make a `SometimesLogger` which implements ITelemetryLogger and also takes another ITelemetryLogger in its constructor. It delegates all operations to the inner logger, but only fires them some percentage of the time. If you passed one of those to the IdCompressor instead, most of the changes in this PR would go away.

Reply:
Gotcha. That's okay. Instead of making it a generic tool, just make it the `IdCompressorLogger`. Even if it's a very specific logger that's only used for the IdCompressor, the real benefit is not having to plumb a new variable and checks all over the IdCompressor.

Reply:
Added `IdCompressorLogger` in latest commit